rgw/rgw.go

```diff
@@ -0,0 +1,805 @@
+package rgw
+
+/*
+#cgo LDFLAGS: -lrgw
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <rados/librgw.h>
+#include <rados/rgw_file.h>
+
+// readdir_callback.go
+extern bool common_readdir_cb(const char *name, void *arg, uint64_t offset,
+                       struct stat *st, uint32_t mask,
+                       uint32_t flags);
+*/
+import "C"
+
+import (
+	"syscall"
+	"unsafe"
+
+	gopointer "github.com/mattn/go-pointer"
```

Comment:
We have our own internal callback helper library under `github.com/ceph/go-ceph/internal/callbacks`. Examples can be found in `rbd/diff_iterate.go` and `rbd/watchers_mimic.go`. If you have any other questions about that callback helper library please ask, but it is requested that you use that rather than rely on a different external dependency.